File: 字符串/1.比较版本号/python/main.py
from functools import cmp_to_key
from typing import List


# import sys
#
# sys.setrecursionlimit(100000)  # 例如这里设置为十万


class Solution:

    def compareVersion(self, version1: str, version2: str) -> int:
        ss1 = version1.split('.')
        ss2 = version2.split('.')
        while len(ss1) > 0 and int(ss1[-1]) == 0:
            ss1.pop(-1)
        while len(ss2) > 0 and int(ss2[-1]) == 0:
            ss2.pop(-1)
        i = 0
        j = 0
        while i < len(ss1) or j < len(ss2):
            if i >= len(ss1):
                return -1
            if j >= len(ss2):
                return 1
            if int(ss1[i]) > int(ss2[j]):
                return 1
            elif int(ss1[i]) < int(ss2[j]):
                return -1
            i += 1
            j += 1
        return 0


if __name__ == '__main__':
    s = Solution()
    s.compareVersion("1.0","1.0.0")

File: 字符串/1.比较版本号/java/Main.java
import java.util.*;

// 1:无需package
// 2: 类名必须Main, 不可修改
public class Main {
    public int compareVersion(String version1, String version2) {
        String[] ss1 = version1.split("\\.");
        String[] ss2 = version2.split("\\.");
        int i = 0, j = 0;
        int index1 = ss1.length - 1;
        int index2 = ss2.length - 1;
        while (index1 >= 0) {
            if (Integer.parseInt(ss1[index1]) == 0) {
                index1--;
            } else {
                break;
            }
        }
        while (index2 >= 0) {
            if (Integer.parseInt(ss2[index2]) == 0) {
                index2--;
            } else {
                break;
            }
        }
        while (i <= index1 || j <= index2) {
            if (i > index1) {
                return -1;
            }
            if (j > index2) {
                return 1;
            }
            int val1 = Integer.parseInt(ss1[i]);
            int val2 = Integer.parseInt(ss2[j]);
            if (val1 < val2) {
                return -1;
            } else if (val1 > val2) {
                return 1;
            }
            i++;
            j++;
        }
        return 0;
    }

    public static void main(String[] args) {
        Main main = new Main();
        main.compareVersion("1", "0");
    }
}